src/docs/resources/architectural-overview.md

```diff
@@ -0,0 +1,966 @@
+typically occurs using the platform channels mechanism, as previously described.
+
+At present, platform views aren’t available for desktop platforms, but this is
+not an architectural limitation; support might be added in the future.
+
+### Hosting Flutter content in a parent app
+
+The converse of the preceding scenario is embedding a Flutter widget in an
+existing Android or iOS app. As described in an earlier section, a newly created
+Flutter app running on a mobile device is hosted in an Android activity or iOS
+`UIViewController`. Flutter content can be embedded into an existing Android or
+iOS app using the same embedding API.
+
+The Flutter module template is designed for easy embedding; you can either embed
+it as a source dependency into an existing Gradle or Xcode build definition, or
+you can compile it into an Android Archive or iOS Framework binary for use
+without requiring every developer to have Flutter installed.
+
+The Flutter engine takes a short while to initialize, because it needs to load
+Flutter shared libraries, initialize the Dart runtime, create and run a Dart
+isolate, and attach a rendering surface to the UI. To minimize any UI delays
+when presenting Flutter content, it’s best to initialize the Flutter engine
+during the overall app initialization sequence, or at least ahead of the first
+Flutter screen, so that users don’t experience a sudden pause while the first
+Flutter code is loaded. In addition, separating the Flutter engine allows it to
+be reused across multiple Flutter screens and share the memory overhead involved
+with loading the necessary libraries.
+
+More information about how Flutter is loaded into an existing Android or iOS app
+can be found at the [Load sequence, performance and memory
+topic](https://flutter.dev/docs/development/add-to-app/performance).
+
+## Flutter web support
+
+While the general architectural concepts apply to all platforms that Flutter
+supports, there are some unique characteristics of Flutter’s web support that
+are worthy of comment.
+
+Dart has been compiling to JavaScript for as long as the language has existed,
+with a toolchain optimized for both development and production purposes. Many
+important apps compile from Dart to JavaScript and run in production today,
+including the [advertiser tooling for Google Ads](https://ads.google.com/home/).
+Because the Flutter framework is written in Dart, compiling it to JavaScript was
+relatively straightforward.
+
+However, the Flutter engine, written in C++, is designed to interface with the
+underlying operating system rather than a web browser. A different approach is
+therefore required. On the web, Flutter provides a reimplementation of the
+engine on top of standard browser APIs. We currently have two options for
+rendering Flutter content on the web: HTML and WebGL. In HTML mode, Flutter uses
+HTML, CSS, Canvas, and SVG. To render to WebGL, Flutter uses a version of Skia
+compiled to WebAssembly called
```

Comment:
CanvasKit isn't ready yet, right?

Reply:
We're starting to cautiously promote it. See, for example: https://medium.com/flutter/going-deeper-with-flutters-web-support-66d7ad95eb52

It's rapidly progressing, and in the next month or so it will become the default.